modules/workspace/main.tf

```diff
@@ -8,7 +8,7 @@ resource "tfe_workspace" "workspace" {
   vcs_repo {
     identifier     = var.repository
     oauth_token_id = var.github_oauth_token_id
-    branch         = "leo/restructure"
+    branch         = coalesce(var.github_workspace_branch, var.TFC_CONFIGURATION_VERSION_GIT_BRANCH, "leo/restructure")
```

Comment:
Why?

Reply:
@ldb and I discussed this today:

My idea was, that a pull request will also use code on it's branch for the subworkspaces. We agreed that this would be a nice feature, however `coalesce` isn't required here. We just use the TFC env variable, the provider falls back to the repositories default branch if not set.